internal/pkg/agent/application/upgrade/upgrade.go

```diff
@@ -63,7 +63,7 @@ type Upgrader struct {
 func IsUpgradeable() bool {
 	// only upgradeable if running from Agent installer and running under the
 	// control of the system supervisor (or built specifically with upgrading enabled)
-	return release.Upgradeable() || (paths.RunningInstalled() && info.RunningUnderSupervisor())
+	return release.Upgradeable() || info.RunningUnderSupervisor()
```

Comment:
We don't allow upgradable unless installed from the `elastic-agent install`. This will break this check. This means that even RPM and DEB's can be self-upgraded, that is not true.

Reply:
Ah I wasn't sure which case this was covering and you are right it will allow RPM and DEB to appear upgradable through Fleet.

We need to put something back back to prevent support cases coming in about this when the upgrade process doesn't work for these installs.

I still don't really like the idea of the entire upgrade process being disabled by deleting a single file. I wonder if we shouldn't put the special case in the RPM and DEB instead?

Is there a way a running agent can detect that it was installed via a DEB or RPM package directly, instead of us inferring it via the presence of a magic file?

Reply:
Yah, removing that was @cmacknz 's idea, since the idea (i think) is that `RunningUnderSupervisor` will be functionally the same as `RunningInstalled`. Is there a case where `RunningUnderSupervisor` will be true but `RunningInstalled` won't be? 

Does the RPM/DEB install not place the `.installed` file?

Reply:
> Does the RPM/DEB install not place the .installed file?

It doesn't since `elastic-agent install` is never invoked. The installation is done by the `dpkg` and `rpm` commands.

I ran `AGENT_PACKAGE_VERSION=8.12.0 EXTERNAL=true SNAPSHOT=true PLATFORMS=linux/arm64 PACKAGES=deb mage package`, installed it in a Linux VM, enrolled it with an 8.13.0-SNAPSHOT Fleet instance, and confirmed Fleet shows it as upgradable.

Reply:
Easiest thing to do now is to put the `Installed` check along with a comment about what its for. We can then evaluate if just creating the installed marker file earlier solves all problems related to upgrade.

Reworking how we detect we are installed on RPM and DEB separately from `elastic-agent install` is out of scope of this PR I think.

Reply:
@cmacknz to be clear, are you saying we should revert the placement of the `.installed` file, and just have this PR fix the error string and add some docs?